cmd/stack_tag.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/pulumi/pulumi/pkg/backend"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+)
+
+func newStackTagCmd() *cobra.Command {
+	var stack string
+
+	cmd := &cobra.Command{
+		Use:   "tag",
+		Short: "Manage stack tags",
+		Long: "Manage stack tags\n" +
+			"\n" +
+			"Stacks have associated metadata in the form of tags. Each tag consists of a name\n" +
+			"and value. The `get`, `ls`, `rm`, and `set` commands can be used to manage tags.\n" +
+			"Some tags are automatically assigned based on the environment each time a stack\n" +
+			"is updated.\n",
+		Args: cmdutil.NoArgs,
+	}
+
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "", "The name of the stack to operate on. Defaults to the current stack")
+
+	cmd.AddCommand(newStackTagGetCmd(&stack))
+	cmd.AddCommand(newStackTagLsCmd(&stack))
+	cmd.AddCommand(newStackTagRmCmd(&stack))
+	cmd.AddCommand(newStackTagSetCmd(&stack))
+
+	return cmd
+}
+
+func newStackTagGetCmd(stack *string) *cobra.Command {
+	return &cobra.Command{
+		Use:   "get <name>",
+		Short: "Get a single stack tag value",
+		Args:  cmdutil.SpecificArgs([]string{"name"}),
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			name := args[0]
+
+			opts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+			s, err := requireStack(*stack, false, opts, true /*setCurrent*/)
+			if err != nil {
+				return err
+			}
+
+			tags, err := backend.GetStackTags(commandContext(), s)
+			if err != nil {
+				return err
+			}
+
+			if value, ok := tags[name]; ok {
+				fmt.Printf("%v\n", value)
+				return nil
+			}
+
+			return errors.Errorf(
+				"stack tag '%s' not found for stack '%s'", name, s.Ref())
+		}),
+	}
+}
+
+func newStackTagLsCmd(stack *string) *cobra.Command {
```

Comment:
I'm on the fence about the `pulumi stack tag ls` command. `pulumi config` lists the config, so we could make `pulumi stack tag` do the same thing and not have an `ls` command. On the other hand, we have `pulumi plugin ls` and `pulumi stack ls` so...

Reply:
I think being explicit and having the `ls` here is good enough.